src/states/elder/mod.rs

```diff
@@ -1202,6 +1192,32 @@ impl Elder {
         self.send_routing_message(src, dst, content)
     }
 
+    fn update_our_knowledge(&mut self, signed_msg: &SignedRoutingMessage) {
+        if signed_msg.routing_message().src.is_client() {
+            return;
+        }
+
+        let sec_info = if let Some(si) = signed_msg.source_section() {
+            si
+        } else {
+            return;
+        };
+
+        let new_key_info = self.chain.get_their_keys_info().any(|(_, info)| {
```

Comment:
just to confirm that is this `any` check correct?

Reply:
In what way would it be not correct?

Reply:
Well, I gave it a second thought, and cann't raise a concrete scenario of failue.
So it's just my personal nervous against the `any newer version of compatible their keys`   :)

Reply:
I think this is what we want, it is similar to `is_new_neighbour`.
At this point this is a trusted message, we could actually remove the condition all together, but we would end up doing this filtering in Parsec (where we would see we already voted for it).